src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -308,7 +308,7 @@ public static MediaTypeSegmentWithQuality CreateMediaTypeSegmentWithQuality(stri
 
             // We check if the parsed media type has value at this stage when we have iterated
             // over all the parameters and we know if the parsing was sucessful.
-            if (!parser.ParsingFailed)
+            if (!parser.ParsingFailed && parser.CurrentOffset >= start)
```

Comment:
Further, what exactly causes `CurrentOffset` to decrease? If that's impossible, shouldn't the new part of this condition be done before the `ParseNextParameter()` loop above?